doc/source/examples/multiphysics/rayleigh-benard-convection/rayleigh-benard-convection.rst

```diff
@@ -0,0 +1,181 @@
+Location of the example
+------------------------
+``examples/multiphysics/rayleigh_benard_convection/rayleigh_benard_convection.prm``
+
+
+-----------------------------
+Description of the case
+-----------------------------
+
+In this example, we evaluate the performance of the ``gls_navier_stokes_2d`` solver in the simulation of the stability of natural convection within a two-dimensional rectangular domain. The following schematic describes the geometry and dimensions of the simulation in the :math:`(x,y)` plane:
+
+.. image:: images/geometry.png
+    :alt: Schematic
+    :align: center
+    :width: 400
+
+
+The incompressible Navier-Stokes equations with a Boussinesq approximation for the buoyant force are:
+    .. math::
+        \nabla \cdot {\bf{u}} = 0
+
+    .. math::
+        \rho \frac{\partial {\bf{u}}}{\partial t} + \rho ({\bf{u}} \cdot \nabla) {\bf{u}} = -\nabla p + \nabla \cdot {\bf{\tau}} - \rho \beta {\bf{g}} (T - T_0)
+
+where :math:`\beta` and :math:`T_0` denote thermal expansion coefficient and a reference temperature, respectively.
+
+A two-dimensional block of fluid is heated from its bottom wall at :math:`t = 0` s. The temperature of the bottom wall is equal to math:`T_h=50`, the temperature of the top wall is equal to math:`T_c=0`, and the left and right walls are insulated. By heating the fluid from the bottom wall, the buoyant force (natural convection) creates vortices inside the fluid. The shape and number of these vortices mainly depend on the Rayleigh number [1, 2]:
+
+    .. math::
+        \text{Ra} = \frac{\rho^2 \beta g (T_h - T_c) H^3}{k \mu}
+
+
+where :math:`\rho` is the fluid density, :math:`\beta` denotes the thermal expansion coefficient, :math:`g` is the magnitude of gravitational acceleration, :math:`T_h` and :math:`T_c` are the hot wall and the cold wall temperatures, :math:`H` denotes the characteristic length, :math:`k` is the thermal conduction coefficient, and :math:`\mu` and :math:`\nu` are the dynamic and kinematic viscosities, :math:`c_p` is the specific thermal capacity, and :math:`\lambda` is the enthalpy of melting.
+
+In this example, we simulate the Rayleigh-Benard convection problem at two Rayleigh numbers of 10000 and 25000. According to the literature [1, 2], we should see different numbers (2 and 3 vortices at Ra=10000 and 25000, respectively) of vortices in the fluid at these two Rayleigh numbers. Note that in this example, we change the Ra number by changing the thermal conduction coefficient of the fluid.
+
+.. note:: 
+    All the four boundary conditions are ``noslip``, and an external 
+    gravity field of :math:`-10` is applied in the :math:`y` direction.
+
+
+--------------
+Parameter file
+--------------
+
+Time integration is handled by a 1st order backward differentiation scheme 
+`(bdf1)`, for a :math:`10000` s simulation time with an initial 
+time step of :math:`0.01` second.
+
+.. note::   
+    This example uses an adaptive time-stepping method, where the 
```

Comment:
Maybe it would be interesting to explain why an adaptive time-stepping method is used.

Reply:
This has been explained in incompressible flow examples.

Reply:
Yes you're right, i didn't see it. I would maybe add the reference to the other example in that case.

```suggestion
    As explained in `2D Flow past a backward-facing step <https://lethe-cfd.github.io/lethe/examples/incompressible-flow/2d-backward-facing-step/2d-backward-facing-step.html>`_, this example uses an adaptive time-stepping method, where the 
```